go/libraries/doltcore/sqle/index/dolt_index.go

```diff
@@ -1230,17 +1230,21 @@ func (di *doltIndex) prollyRangesFromSqlRanges(ctx context.Context, ns tree.Node
 		order := di.keyBld.Desc.Comparator()
 		for i, field := range fields {
 			// lookups on non-unique indexes can't be point lookups
+			typ := di.keyBld.Desc.Types[i]
+			cmp := order.CompareValues(i, field.Hi.Value, field.Lo.Value, typ)
+			fields[i].Exact = cmp == 0
+
 			if !di.unique {
-				fields[i].Exact = false
+				fields[i].Unique = false
 				continue
 			}
 			if !field.Hi.Binding || !field.Lo.Binding {
+				// infinity bound
+				fields[i].Unique = false
 				fields[i].Exact = false
 				continue
 			}
-			typ := di.keyBld.Desc.Types[i]
-			cmp := order.CompareValues(i, field.Hi.Value, field.Lo.Value, typ)
-			fields[i].Exact = cmp == 0
+			fields[i].Unique = cmp == 0
```

Comment:
This doesn't seem right, and indicates that these two properties (unique, exact) are being conflated in different circumstances